### PR TITLE
randomise user agent to avoid rate limit blocking

### PR DESCRIPTION
**Pull request by vasuki1996**

### CODE REVIEW

**Comment by Nakul93** [comment]
Have you tested this?

File: src/ratelimit.py [hunk 1]
```diff
@@ -1,8 +1,28 @@
...
 def handle_rate_limited():
     print('Rate-limited by CoWIN. Waiting for 5 seconds.\n'
           '(You can reduce your refresh frequency. Please note that other devices/browsers '
           'using CoWIN/Umang/Arogya Setu also contribute to same limit.)')
     time.sleep(5)
+
+def randomiseAgentToAvoidRateLimit():
+    base_headers_array = [
+        {
+            'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36',
```
Comment (BlackRider97) [inline src/ratelimit.py:13]: why not random number here ?
10_{random}_1

Is it being validated ?

**Comment by bombardier-gif** [comment]
does this work?

File: src/ratelimit.py (hunk 1 above)
Comment (vasuki1996) [reply src/ratelimit.py:13]: These are 3 valid variants I have tried as of now, I'm trying out other variations as well

**Comment by vasuki1996** [comment]
> does this work?

Yes, I have been testing this for last 3 days

**Comment by Nakul93** [comment]
> > does this work?
> 
> Yes, I have been testing this for last 3 days

And how long was the script able to run without rate limiting error

**Comment by vasuki1996** [comment]
almost 3 hours, but the slots get booked by then

**Comment by bombardier-gif** [comment]
Hi @vasuki1996 Can you confirm that you ran it with 1 second frequency and it worked fine?
If yes then its great, if not then we will have to test. I can help you test this

**Comment by dhhruv** [comment]
Why not use [https://pypi.org/project/fake-useragent/](https://pypi.org/project/fake-useragent/) to get a random UserAgent?

**Comment by shailesh** [comment]
This doesn't work. Had raised the issue a month back. #185

**Comment by vasuki1996** [comment]
> Hi @vasuki1996 Can you confirm that you ran it with a 1-second frequency and it worked fine?
> If yes then it's great, if not then we will have to test. I can help you test this

@bombardier-gif I ran it with 5-second frequency, will run it on 1-second frequency

**Comment by bombardier-gif** [comment]
I also tried to work on ways to bypass the rate-limiter. 
Based on the evidence that cowin uses AWS, the rate limiter seems to be on the connection IP address.

**Comment by shailesh** [comment]
> I also tried to work on ways to bypass the rate-limiter.
> Based on the evidence that cowin uses AWS, the rate limiter seems to be on the connection IP address.

Right

**Comment by vasuki1996** [comment]
I ran a couple of tests from my side to confirm this, I don't think so they are using AWS API Gateway. I'm currently working on projects which integrate AWS API Gateway and none of them unblock themselves by change in User-Agent. These are my observations, completely open for corrections

**Comment by shailesh** [comment]
We can abandon this PR for now. We are sure and tested that this doesn't fix the rate limiting issue. @bombardier-gif

**Comment by 6mxo89sqh** [comment]
Closing this for now as @shailesh tried it and it doesn't work.

**Pull request closed without merging**